src/PROPOSAL/geometry/Box.cxx

```diff
@@ -95,9 +94,9 @@ std::pair<double, double> Box::DistanceToBorder(const Vector3D& position, const
     double z_calc_neg = position_.GetZ() - 0.5 * z_;
 
     // intersection with E1
-    if (dir_vec_x != 0) // if dir_vec == 0 particle trajectory is parallel to E1
+    if (dir_vec[0] != 0) // if dir_vec == 0 particle trajectory is parallel to E1
```

Comment:
I think `dir_vec.GetX()` would be more readable. Same for the following lines

Reply:
This is a good idea. `dir_vec` is actually an `std::array `object at the moment, but the overhead of creating a Cartesian3D object and using its Getter functions instead should be minimal (or even zero). However, this should increase the readability and ensure that one does not confuse Spherical3D and Cartesian3D objects (since using the [] operator instead of Getter functions will not indicate when you are confusing the object you are working with at the moment).